st-app-frontend/src/app/api/auth/github/token/route.ts

```diff
@@ -0,0 +1,46 @@
+import { NextRequest, NextResponse } from "next/server";
+
+export async function POST(request: NextRequest) {
+  try {
+    const { code } = await request.json();
+
+    if (!code) {
+      return NextResponse.json({ error: "Code is required" }, { status: 400 });
+    }
+
+    const params = new URLSearchParams({
+      client_id: process.env.NEXT_PUBLIC_GITHUB_CLIENT_ID!,
```

Comment:
It would be better to check env variables before forming the request, and send a 500 if env variables are missing